[v1.x][F09-B4] Vetter Appointments

md-azsa opened the pull request:
Initial P.R
@jonathanwj 
@Godxin 
@purplepers0n

k-l-a commented on src/main/java/seedu/address/logic/commands/AddPetCommand.java:
```diff
@@ -0,0 +1,61 @@
+import seedu.address.model.pet.exceptions.DuplicatePetException;
+
+/**
+ * Adds support for adding a pet into the program.
```
This should just describe what this command does. (e.g. `Adds a pet into the application.`)

k-l-a reviewed (commented):
Code seems fine. Please update UG with documentation of all features up to V2.0, and mark them accordingly. This was required for last week's team component, so I will have to mark it as missed.

Also, please update the title of this PR to reflect your product version each time you release a new version. (e.g. [v1.1])

@md-azsa @jonathanwj @Godxin @purplepers0n 
In addition, I would want you to mention the following in your **Developer guide**
1. For each team member what are the features (major and minor) you are proposing?
2. Within the scope of the project, how does it fit in (just a one or 2 line summary)
Please do this by end of day Monday 19 March.

jonathanwj committed:
Add support for autocomplete for parameters

purplepers0n committed:
Set up UI java files to display appt from list

jonathanwj committed:
AutoComplete: Update JUnit tests

jonathanwj committed:
Add AddVetTechnToAppointmentCommand class

md-azsa committed:
Sortp command now throws a ClientpetAssociationListEmptyException when trying to sort an empty list.

jonathanwj committed:
AddVetTechToAppointmentCommand: Add new command class

md-azsa committed:
Adds test cases and methods for SortPetCommand

purplepers0n committed:
schedule command to update appt list

jonathanwj committed:
AddVetTechToAppointment: Update JUnit tests

jonathanwj committed:
Add RemoveVetTechFromAppointmentCommand class

Godxin committed:
add reschedule builder

Godxin committed:
implements updateappt in modelmanager

Godxin committed:
add update aapt in model

Godxin committed:
add update appt in addressbook

Godxin committed:
add some in parseUtil

Godxin committed:
add reschedule in addressbook parser

jonathanwj committed:
RemoveVetTechFromAppointment: Update JUnit tests

jonathanwj committed:
Merge branch 'remove-vet-from-appt' into autocomple-fix

purplepers0n committed:
first pass of appointment display setup

jonathanwj committed:
AutoComplete: Add new commands

jonathanwj committed:
AutoComplete: Add new commands

jonathanwj committed:
AutoComplete: Add more commands

jonathanwj committed:
AutoComplete: Add more commands

jonathanwj committed:
AutoComplet: Fix Junit tab test

jonathanwj committed:
AutoComplet: Fix Junit tab test

jonathanwj committed:
Collate annotate

jonathanwj committed:
Remove accidentally added

jonathanwj committed:
Remove accidentally added

jonathanwj committed:
Merge branch 'autocomple-fix' into collate

jonathanwj committed:
Collate Shift unused code to unused folder

jonathanwj committed:
Collate output

jonathanwj committed:
Merge branch 'upstream-master' into branch

# Conflicts:
#	src/main/java/seedu/address/logic/parser/AddressBookParser.java
#	src/main/java/seedu/address/model/AddressBook.java
#	src/main/java/seedu/address/model/Model.java
#	src/main/java/seedu/address/model/ModelManager.java
#	src/main/java/seedu/address/model/appointment/Appointment.java
#	src/test/java/seedu/address/logic/commands/AddCommandTest.java
#	src/test/java/seedu/address/logic/commands/AddPetCommandTest.java
#	src/test/java/seedu/address/logic/commands/ScheduleCommandTest.java
#	src/test/java/seedu/address/logic/parser/AddressBookParserTest.java

jonathanwj committed:
Merge pull request #160 from jonathanwj/remove-vet-from-appt

Add RemoveVetTechFromApppointmentCommand

jonathanwj committed:
Merge branch 'upstream-master' into branch

# Conflicts:
#	src/main/java/seedu/address/logic/autocomplete/CommandParameterSyntaxHandler.java
#	src/main/java/seedu/address/ui/CommandBox.java
#	src/test/java/seedu/address/logic/autocomplete/CommandParameterSyntaxHandlerTest.java

jonathanwj committed:
AddTests for petclientassociation

jonathanwj committed:
Add Tests

jonathanwj committed:
Merge pull request #161 from jonathanwj/collate

Collate jonathanwj

md-azsa committed:
Ammends the comments according to @jonathanwj

md-azsa committed:
Adds collab

md-azsa committed:
collated

purplepers0n committed:
display correct index for appointments

md-azsa committed:
Adds test for XmlAdaptedPet

md-azsa committed:
Adds test cases for UniquePetList

md-azsa committed:
Adds support so that SortClient list will throw an exception when called upon on empty list

md-azsa committed:
Adds test cases for SortClientCommand

purplepers0n committed:
change label colours

purplepers0n committed:
refresh page when appt added to pet

purplepers0n committed:
update design

purplepers0n committed:
add newline at eof

purplepers0n committed:
revert additional line

purplepers0n committed:
remove unused import

purplepers0n committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B4/main

Godxin committed:
update apptbuilder

Godxin committed:
add appt in cmdtestutil

Godxin committed:
add reschedule cmd test

Godxin committed:
improve reschedule cmd

Godxin committed:
add collate tag

purplepers0n committed:
fix exception when appointment list is empty

purplepers0n committed:
update UI images

jonathanwj committed:
AddressBook#removePerson(): Add support for removing vet technician from existing appointments when removed.

jonathanwj committed:
Fix bugs related to adding/removing vetTech to/from appointment

- Add vetTech to XmlAdaptedAppointment
- Add raise apptcardeventchanged on addressbook redo and undo
- Add vettech not found when removing vetech from appt

jonathanwj committed:
Add execption clause

md-azsa committed:
Deleting a client now deletes all pets and association in list (#167)

* Deleting a client will now delete all of it's association and pets in all three lists

* Changed the comment to better suit the changes.

* fixed conflicts

md-azsa committed:
ApptListPanel now grows according to window size

md-azsa committed:
More visible headers

md-azsa committed:
Changed to font used in landing page

Godxin committed:
rescheduleCommand: fix mistake

Godxin committed:
fix implementation issue

Godxin committed:
testUtil: update

Godxin committed:
checkstyle fix

Godxin committed:
fix implementation issue

Godxin committed:
addressbookParserTest: upate

Godxin committed:
add collate tag

Godxin committed:
Merge branch 'master' into rescheduleCommand

Godxin committed:
fix bug

Godxin committed:
checkstyle fix

Godxin committed:
bug fix

Godxin committed:
Merge pull request #171 from Godxin/rescheduleCommand

Add Reschedulecommand to edit details of an appointment

Godxin committed:
add reschedule command

Godxin committed:
add collate tag

Godxin committed:
collate file update

md-azsa committed:
Changed the Ui.PNG

md-azsa committed:
Collate

Godxin committed:
Merge branch 'master' into UGupdateV1.4

Godxin committed:
Merge pull request #173 from Godxin/UGupdateV1.4

Update UserGuide with rescheduleCommand

purplepers0n committed:
Merge branch 'master' of https://github.com/CS2103JAN2018-F09-B4/main

jonathanwj committed:
UserGuide: Update autocomplete component

jonathanwj committed:
UserGuide: Update minor errors

jonathanwj committed:
FindCommand: Fix bug

jonathanwj committed:
Fix johndoe.adoc Offset

jonathanwj committed:
Remove unused variable from Client

purplepers0n committed:
update logic and model

purplepers0n committed:
update dg

purplepers0n committed:
Create listall command

jonathanwj committed:
UserGuide: Add removevettechfromappt and addvettechtoappt

md-azsa committed:
UserGuide.adoc update until pets

md-azsa committed:
So tired from doing userguide

jonathanwj committed:
Merge branch 'upstream-master' into PR

# Conflicts:
#	docs/UserGuide.adoc

jonathanwj committed:
UG: Remove select command

jonathanwj committed:
Merge pull request #179 from jonathanwj/2fix-ugdg

Update UG

md-azsa committed:
Now deletes everything the client has in the data file
Adds new sample data

md-azsa committed:
Fixes issue #185

md-azsa committed:
Fixes appt bug

jonathanwj committed:
Add tests for list and find

purplepers0n committed:
Add event to refresh UI when appointment is rescheduled

jonathanwj committed:
RemoveApptFromPet: Fix update appt list on remove

purplepers0n committed:
remove "pet owner: " string that displays nothing

purplepers0n committed:
Allow for appointment splitpane to be resizable

purplepers0n committed:
call event in model instead of individual commands

purplepers0n committed:
remove unused imports

jonathanwj committed:
AddressBook#removePet(): remove pet from all appointments that has the specified pet.

jonathanwj committed:
Add Tests for removePet

jonathanwj committed:
SortCommands: Fix comparison bug

purplepers0n committed:
first pass of update

jonathanwj committed:
UserGuide: Fix addapptopet command format typo

jonathanwj committed:
AutoComplete: Refactor code

jonathanwj committed:
DG: Add AutoComplete

jonathanwj committed:
DG: fix typo

purplepers0n committed:
update UI class diagram

jonathanwj committed:
AutoComplete: Junit Update Tests

purplepers0n committed:
Remove browser panel in diagram

jonathanwj committed:
DG: Fix trailing whitespace

jonathanwj committed:
DG: Remove AddPetToClient

purplepers0n committed:
update DG

purplepers0n committed:
remove old implementation

jonathanwj committed:
DG: Minor changes

jonathanwj committed:
Merge branch 'upstream-master' into ugdg

# Conflicts:
#	docs/DeveloperGuide.adoc

jonathanwj committed:
Fix Merge

jonathanwj committed:
DG: Update

jonathanwj committed:
DG: fix typo

jonathanwj committed:
Merge pull request #210 from jonathanwj/ugdg

Ugdg update

jonathanwj committed:
UG: Fix removeapptfrompet typo

jonathanwj committed:
UG: Update summary table

jonathanwj committed:
UG: Reschedule typo

jonathanwj committed:
UG: Update

purplepers0n committed:
update portfolio page (first pass)

Godxin committed:
UG: add some TIP for schedule command

purplepers0n committed:
Finish listall command execution

purplepers0n committed:
add author to collate code

Godxin committed:
UG: add pic for reschedule command

purplepers0n committed:
fix typo

purplepers0n committed:
setup listallpanel ui file

purplepers0n committed:
change appt styling

purplepers0n committed:
change appt display to dark

purplepers0n committed:
for client display in listallpanel section (like client card but without index)

purplepers0n committed:
for appt display in listallpanel section (same layout format as client/pet cards)

purplepers0n committed:
for pet display in listallpanel section (like pet card but without client)

purplepers0n committed:
fix

purplepers0n committed:
some setup

purplepers0n committed:
add events

purplepers0n committed:
command to call event

purplepers0n committed:
update mainwindow

purplepers0n committed:
remove selection

purplepers0n committed:
Set listall display on UI

purplepers0n committed:
update existing tests

purplepers0n committed:
adjust display

purplepers0n committed:
Merge branch 'master' into listall

purplepers0n committed:
accomodate to modelmanager method name change

purplepers0n committed:
minor update

purplepers0n committed:
fix checkstyle

purplepers0n committed:
Merge pull request #231 from purplepers0n/listall

Listall command

purplepers0n committed:
update ug for listall

md-azsa committed:
Unschedule appointment (#233)

* Adds the UnscheduleCommand

* Adds support for unschedule command.

* Adds the UnscheduleCommandParser

* Removes the SortAppointmentCommand

* Supports unschedule command

* Fixed some of the test case

* Updates the UserGuide.adoc for the unschedulecommand.

* Updates the summarized tables

* Adds test cases for UnscheduleCommand

* Adds more test cases

* no message

* Adds a test case for UnscheduleCommandParser

Godxin committed:
Fix bugs in validating date  (#230)

* UG: add some TIP for schedule command

* UG: add pic for reschedule command

* bug fix: date-year

* add test for date

* parseUtil test update

jonathanwj committed:
Remove select command usage

Godxin committed:
date: add checks for leap year and days in month

Godxin committed:
update parserutil for date

Godxin committed:
update test

jonathanwj committed:
Remove selection events from panels

purplepers0n committed:
create basic tests for listall command

purplepers0n committed:
fix typos

purplepers0n committed:
listall panel will be cleared if the queried client is deleted or AB is cleared

purplepers0n committed:
remove index

purplepers0n committed:
add tests to ensure listallpanel is cleared upon AB clear or client delete

purplepers0n committed:
deleting other client that is not on listall panel should not change the panel

Godxin committed:
UG: remove unnecessary info for schedule and reschedule

purplepers0n committed:
collated codes

purplepers0n committed:
remove unused import

purplepers0n committed:
fix

purplepers0n committed:
edit PPP

purplepers0n committed:
update DG

md-azsa committed:
Documentation Update (#241)

* Adds PPP page

* User guide update

* Followed feedback given

purplepers0n committed:
Update UG

purplepers0n committed:
update PPP

md-azsa committed:
Az dg update/ppp (#242)

* Adds PPP page

* User guide update

* Followed feedback given

* no message

* no message

* no message

* no message

* no message

* no message

* no message

* no message

* DG Update

* no message

md-azsa committed:
Azhar PPP (#243)

* Adds PPP page

* User guide update

* Followed feedback given

* Add tags to UG for reference

* PPP Update

md-azsa committed:
Update AboutUs.adoc

purplepers0n committed:
add tag

purplepers0n committed:
update ug

purplepers0n committed:
update aboutus

purplepers0n committed:
update ppp

md-azsa committed:
reduce pages ppp (#245)


* Reduce amount of pages

Godxin committed:
UG upate: schedule & reschrdule

Godxin committed:
DG:  appointment related update

jonathanwj committed:
Remove unused events

jonathanwj committed:
Fix close on minimize

jonathanwj committed:
Fix Minimize window close bug

jonathanwj committed:
Add more commands to autocomplete

jonathanwj committed:
CommandBox: AddShortcuts

purplepers0n committed:
fix clear listallpanel

jonathanwj committed:
UserGuide: Update UG

jonathanwj committed:
UG: remove line break

jonathanwj committed:
UserGuide: Change minor details

jonathanwj committed:
Update UG

jonathanwj committed:
Update UG

Godxin committed:
DG: update

jonathanwj committed:
DG: add prefaces

Godxin committed:
Merge pull request #254 from Godxin/UGDUv1.5.2

UGDG update: appointment related

Godxin committed:
add tags for PPP

Godxin committed:
zhengxin ppp

jonathanwj committed:
JonathanWeng.adoc: Add portfolio

jonathanwj committed:
Add new prefixes to autocomplete reschedule

jonathanwj committed:
Autocomplete: Add test

Godxin committed:
add client and vettch

Godxin committed:
fix reschedule: keep clientownpet and vettech

md-azsa committed:
Update AboutUs.adoc

jonathanwj committed:
AddressBook: Fix bugs

jonathanwj committed:
SampleData: Add more data

jonathanwj committed:
ugdg updates (#262)

* DG: add autocomplete sequence diagram

* About Us: Modify name

* Readme: update url

* DG: Add entity implementation

* Edit

md-azsa committed:
Azhar devguide update (#264)

purplepers0n committed:
update some use cases

purplepers0n committed:
update PPP

purplepers0n committed:
date label to take max width of available space

purplepers0n committed:
update css author

purplepers0n committed:
collated code

purplepers0n committed:
update PPP

jonathanwj committed:
Fix tests for sample data

jonathanwj committed:
Fix Reschdule, Schdule bug

Godxin committed:
fix bug and refactor code

Godxin committed:
add collate tag

Godxin committed:
update test

Godxin committed:
Merge branch 'master' into Moretest

Godxin committed:
collate file update

Godxin committed:
Update AboutUs.adoc

Godxin committed:
Merge pull request #267 from Godxin/Moretest

fix the duration check bug

Godxin committed:
Update UserGuide.adoc

jonathanwj committed:
Update PPP, fix spelling

jonathanwj committed:
fix toc

jonathanwj committed:
Fix table width

Godxin committed:
Update DeveloperGuide.adoc

Godxin committed:
Update ZhongZhengXin-ppp.adoc

Godxin committed:
Update ZhongZhengXin-ppp.adoc

Godxin committed:
Update ZhongZhengXin-ppp.adoc

Godxin committed:
fix pic

Godxin committed:
Update ZhongZhengXin-ppp.adoc

jonathanwj committed:
Fix trailing white space

Godxin committed:
Update ZhongZhengXin-ppp.adoc

jonathanwj committed:
Rename Collate

jonathanwj committed:
PPP update

Godxin committed:
add team html
